common/utils/src/test/scala/org/apache/spark/util/StructuredLoggingSuite.scala

```diff
@@ -228,6 +228,37 @@ trait LoggingSuiteBase
           verifyMsgWithConcat(level, logOutput)
       }
   }
+
+  test("LogEntry should construct MessageWithContext only once") {
```

Comment:
Perhaps we should put the above testing logic into `MDCSuite` or a new UT (eg: `MessageWithContextSuite`), as it seems unrelated to the classification of `structured` and `pattern` logs, so it will be executed twice in GA?

Reply:
A test takes 200ms, running twice is ok. Also, we need to ensure the log entry is constructed only once when structured logging is disabled, too.